src/Data/Argonaut/Decode.purs

```diff
@@ -107,3 +109,30 @@ module Data.Argonaut.Decode
 
   -- arrayMembersL :: forall a. (DecodeJson a, EncodeJson a) => IndexedTraversalP JNumber Json a
   -- arrayMembersL = decodeL >>> traversed >>> arrayL
+
+
+  foreign import json2JSDateImpl """
+  function json2JSDateImpl(json, just, nothing) {
+    if (json && json.constructor && json.constructor == Date) {
```

Comment:
Sorry, I should have looked at this native code more closely before -- How is this a valid JSON encoding (i.e. can survive `parse` / `stringify`)?